enhancements/sig-architecture/30-clusterset-override/README.md

```diff
@@ -0,0 +1,191 @@
+2. Create a managedClusterSet to select all the backup clusters.
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterSet
+metadata:
+name: backup
+spec:
+exclusiveKey: clusterAttribute
+```
+ 
+### I want to create a managedClusterSet to group clusters based on the middleware enabled on these clusters, so I could run special applications in these clusters.
+1. For all clusters which enabled middleware, I need to set label: `middlewareAttribute=middlewareenabled`
+2. Create a managedClusterSet to select all clusters which enabled middleware.
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterSet
+metadata:
+name: middlewareenabled
+spec:
+exclusiveKey: middlewareAttribute
+```
+
+### Risks and Mitigation
+1. If I want to create a managedClusterSet, I must get permissions about two kinds of resources, `managedclustersets/exclusiveKey` and `managedclustersets`. When there are multiple resources for both `managedclustersets/exclusiveKey` and `managedclustersets`, it may cause permission leak.
+ 
+For example:
+a. I want to have permission to create a managedClusterSet which name is `apac` and `exclusiveKey` is `region`, I need to get the following permissions.
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+ name: clustersetrole
+rules:
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets/exclusiveKey"]
+   resourceNames: ["region"]
+   verbs: ["create"]
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets"]
+   resourceNames: ["apac"]
+   verbs: ["*"]
+```
+ 
+b. I want to has permission to create another managedClusterSet which name is `aws` and `exclusiveKey` is `vendor`, I need to get the following permissions
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+ name: clustersetrole
+rules:
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets/exclusiveKey"]
+   resourceNames: ["vendor"]
+   verbs: ["create"]
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets"]
+   resourceNames: ["aws"]
+   verbs: ["*"]
+```
+ 
+c. If I get permissions in both #a and #b, Actually, I also get the permissions to create another two managedClusterSets. This is unexpected.
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterSet
+metadata:
+name: apac
+spec:
+exclusiveKey: vendor
+```
+
+and
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterSet
+metadata:
+name: aws
+spec:
+exclusiveKey: region
+```
+
+2. Currently, their is no permission check when adding a managedcluster to clusterset. User may add a cluster into a clusterset incidentally by adding a label to cluster, which is the exclusiveKey of that clusterset.
+
```

Comment:
I think we need to discuss about mitigation approach here.

Reply:
> I think we need to discuss about mitigation approach here.

mitigation or migration?

Reply:
I agree that describing migration is useful.

1. This could impact clients who expect mutually exclusive sets of managedclusters: this is likely to be small and faced no matter what we do.
2. As long as the old RBAC check for setting `cluster.open-cluster-management.io/clusterset` still functions, old clients changing the label will be ok.  the fact that a new RBAC check could also grant permission won't stop those clients from working.

Any other cases to be concerned about?

Reply:
yes, I mean migration. I think we should also discuss the impact to the consumers during this migration.